src/server/rpc/jsdoc-extractor.js

```diff
@@ -0,0 +1,162 @@
+const fse = require('fs-extra'),
+    Logger = require('../logger.js'),
+    logger = new Logger('netsblox:jsdoc'),
+    doctrine = require('doctrine');
+
+
+const MARKER_START = '/**',
+    MARKER_START_SKIP = '/***',
+    MARKER_END = '*/';
+
+let parseSync = (filePath, searchScope = 5) => {
+    let source = fse.readFileSync(filePath, 'UTF8');
+    return parseSource(source, searchScope);
+};
+
+function parseSource(source, searchScope){
+    let lines = source.split(/\n/);
+    let blocks = extractDocBlocks(source);
+    blocks = blocks.map(block => {
+        let src = block.lines.join('\n');
+        block.parsed = doctrine.parse(src, {unwrap: true});
+        return block;
+    });
+
+    blocks = blocks.filter(block => {
+        let linesToSearch = lines.slice(block.endLine, block.endLine + searchScope);
+        let fnName;
+        // if @name is set jsut use that and save a few cycles
+        if (block.parsed.tags.some(tag => tag.title === 'name')) {
+            fnName  = block.parsed.tags.find(tag => tag.title === 'name').name;
```

Comment:
It isn't a huge deal but you could combine the top two so you aren't searching for the same thing twice.
```
let nameTag = block.parsed.tags.find(tag => tag.title === 'name');
if (nameTag) {
    fnName = nameTag.name;
```